src/wristpy/core/orchestrator.py

```diff
@@ -0,0 +1,266 @@
+            exist.
+    """
+    if not output.parent.exists():
+        raise DirectoryNotFoundError(f"The directory:{output.parent} does not exist.")
+    if output.suffix not in [".csv", ".parquet"]:
+        raise InvalidFileTypeError(
+            f"The extension: {output.suffix} is not supported."
+            "Please save the file as .csv or .parquet",
+        )
+
+
+def format_sleep_data(
+    sleep_windows: List[analytics.SleepWindow], reference_measure: models.Measurement
+) -> np.ndarray:
+    """Formats sleep windows into an array for saving.
+
+    Args:
+        sleep_windows: The list of time stamp pairs indicating periods of sleep.
+        reference_measure: The measure from which the temporal resolution will be taken.
+
+    Returns:
+        1-D np.ndarray, with 1 indicating sleep. Will be of the same length as
+            the timestamps in the reference_measure.
+    """
+    logger.debug("Formatting sleep array from sleep windows.")
+    sleep_array = np.zeros(len(reference_measure.time))
+
+    for window in sleep_windows:
+        sleep_mask = (reference_measure.time >= window.onset) & (
+            reference_measure.time <= window.wakeup
+        )
+        sleep_array[sleep_mask] = 1
+
+    return sleep_array
+
+
+def format_nonwear_data(
+    nonwear_data: models.Measurement,
+    reference_measure: models.Measurement,
+    original_temporal_resolution: float,
+) -> np.ndarray:
+    """Formats nonwear data to match the temporal resolution of the other measures.
+
+    Args:
+        nonwear_data: The nonwear array to be upsampled.
+        reference_measure: The measurement we match the non_wear data's temporal
+            resolution to.
+        original_temporal_resolution: The original temporal resolution of the
+            nonwear_data.
+
+    Returns:
+        1-D np.ndarray with 1 indicating a nonwear timepoint. Will match the
+            length of the reference measure.
+    """
+    logger.debug("Upsampling nonwear data.")
+    nonwear_df = pl.DataFrame(
+        {
+            "nonwear": nonwear_data.measurements.astype(np.int64),
+            "time": nonwear_data.time,
+        }
+    ).set_sorted("time")
+
+    nonwear_array = np.zeros(len(reference_measure.time))
+
+    for row in nonwear_df.iter_rows(named=True):
+        nonwear_value = row["nonwear"]
+        time = row["time"]
+        nonwear_mask = (reference_measure.time >= time) & (
+            reference_measure.time
+            <= time + datetime.timedelta(seconds=original_temporal_resolution)
+        )
+        nonwear_array[nonwear_mask] = nonwear_value
+
+    return nonwear_array
+
+
+def run(
+    input: pathlib.Path,
+    output: Optional[pathlib.Path] = None,
+    settings: config.Settings = config.Settings(),
+    calibrator: Union[
+        None,
+        Literal["ggir", "gradient"],
+    ] = "gradient",
+    epoch_length: Union[int, None] = 5,
+) -> Results:
+    """Runs wristpy.
```

Comment:
Maybe we can be a bit more descriptive? "Runs the main processing steps of wristpy to generate data for analysis of physical activity and sleep data" or something along those lines?

Reply:
good point, added a bit more description.